pkg/apis/core/v1beta1/helper/seed.go

```diff
@@ -105,6 +105,24 @@ func SeedBackupCredentialsRefEqual(oldBackup, newBackup *gardencorev1beta1.Backu
 	return apiequality.Semantic.DeepEqual(oldCredentialsRef, newCredentialsRef)
 }
 
+// InternalDNSProviderEqual returns true when the credentials reference of the internal DNS provider configuration is the same.
```

Comment:
Having in mind that the other fields `type`, `domain` and `zone` are not immutable, shouldn't this function check for change in their values? If not, I would rename the function to `InternalDNSProviderCredentialsRefEqual`.